File: backend/src/main/java/com/auradot/backend/model/Item.java
package com.auradot.backend.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "items")
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "img_url")
    private String imgUrl;

    @ManyToOne
    @JoinColumn(name = "category_id",nullable = true)
    @JsonManagedReference
    private ItemCategory itemCategory;

    @Column(name = "description")
    private String description;

    @Column(name = "price")
    private Float price;

    @Column(name = "is_active", columnDefinition = "TINYINT(1) DEFAULT 1")
    private Boolean isActive = true;

    @OneToOne
    @JsonManagedReference
    private Inventory inventory;

}

File: frontend/src/app/common/item.ts
export class Inventory {
    constructor(
      public stockStatus: string,
      public stockQuantity: number,
      public minimumStockLevel: number
    ) {}
  }
  
  export class Item {
    constructor(
      public name: string,
      public description: string,
      public categoryId: number,
      public imgUrl: string,
      public price: number,
      public inventory: Inventory
    ) {}
  }
  
  export class ItemResponse{
    constructor(
      public  id: number | null,
      public name: string,
      public description: string,
      public imgUrl: string,
      public category: string,
      public stockQuantity: number,
      public minimumStockLevel: number,
      public price: number,
      public created_at: string,
      public status: string){

      }
  }

  export class updateRequest{
    constructor(
      public name: string,
      public description: string,
      public stockQuantity: number,
      public minimumStockLevel: number,
      public price: number
    ){

    }
  }